engine/src/main/java/org/camunda/bpm/engine/impl/util/ExceptionUtil.java

```diff
@@ -181,6 +181,23 @@ public static boolean checkVariableIntegrityViolation(Throwable cause) {
     return false;
   }
 
+  public static Boolean checkCrdbTransactionRetryException(Throwable cause) {
+    List<SQLException> relatedSqlExceptions = findRelatedSqlExceptions(cause);
+    for (SQLException exception : relatedSqlExceptions) {
+      String errorMessage = exception.getMessage().toLowerCase();
+      int errorCode = exception.getErrorCode();
+      if ((errorCode == 40001 || errorMessage != null)
+          && (errorMessage.contains("restart transaction") || errorMessage.contains("retry txn"))
+          // TX retry errors with RETRY_COMMIT_DEADLINE_EXCEEDED are handled
+          // as a ProcessEnginePersistenceException due to a long-running transaction
```

Comment:
There is no class ProcessEnginePersistenceException

Reply:
I clarified the comment. The exception is `ProcessEngineException` (cause: Process engine persistence exception)